### PR TITLE
changes for 2015-* profile to work against azure stack 

### PR DESCRIPTION
**Pull request by bganapa**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by bganapa** [comment]
cc:@derekbekoe

File: src/command_modules/azure-cli-storage/setup.py [hunk 1]
```diff
@@ -24,12 +26,19 @@
     'License :: OSI Approved :: MIT License',
 ]
 
+# Todo: azure-storage package to be updated with cli specific multi versioned package
+if get_api_version(ResourceType.MGMT_STORAGE) in ['2016-12-01']:
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-storage/setup.py]: This won't be needed after https://github.com/Azure/azure-cli/pull/2796.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py [hunk 2]
```diff
@@ -237,36 +243,38 @@ def transform_av_set_collection_output(av_sets):
...
-                           setter_arg_name='snapshot', factory=cf_snapshots)
-
-op_var = 'images_operations'
-op_class = 'ImagesOperations'
-cli_command(__name__, 'image create', custom_path.format('create_image'))
-cli_command(__name__, 'image list', custom_path.format('list_images'))
-cli_command(__name__, 'image show', mgmt_path.format(op_var, op_class, 'get'), cf_images, exception_handler=empty_on_404)
-cli_command(__name__, 'image delete', mgmt_path.format(op_var, op_class, 'delete'), cf_images)
+
+# Todo: add the list of api-versions instead of just one
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: Currently, we support 2015-06-15 and 2016-04-30-preview.
So there's nothing else to add to this list right now.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -664,8 +671,7 @@ def _reset_windows_admin(vm_instance, resource_group_name, username, password, n
     You can only change the password. Adding a new user is not supported.
     '''
     client = _compute_client_factory()
-
-    from azure.mgmt.compute.models import VirtualMachineExtension
+    VirtualMachineExtension = get_versioned_models(ResourceType.MGMT_COMPUTE,"VirtualMachineExtension")
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: style: pylint will likely complain. There should be a space after the `,`

File: src/command_modules/azure-cli-vm/setup.py [hunk 4]
```diff
@@ -24,17 +26,25 @@
     'License :: OSI Approved :: MIT License',
 ]
 
+
+# Todo: azure-storage package to be updated with cli specific multi versioned package
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/setup.py]: Can remove this due to https://github.com/Azure/azure-cli/pull/2796

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 5]
```diff
@@ -716,12 +719,19 @@ def build_vmss_resource(name, naming_prefix, location, tags, overprovision, upgr
...
+    else:
+        vmss_api_version='2016-03-30'
+
     vmss = {
         'type': 'Microsoft.Compute/virtualMachineScaleSets',
         'name': name,
         'location': location,
         'tags': tags,
-        'apiVersion': '2016-04-30-preview',
+        'apiVersion': vmss_api_version,
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Should get the API version from `get_api_version` like you do for `vm_api_version` above.

Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Same with `av_set_api_version` below.

File: src/command_modules/azure-cli-storage/setup.py (hunk 1 above)
Comment (bganapa) [inline src/command_modules/azure-cli-storage/setup.py]: reverted this commit

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py [hunk 6]
```diff
@@ -194,7 +194,10 @@ def _dont_fail_not_exist(ex):
 # queue commands
 factory = queue_data_service_factory
 cli_storage_data_plane_command('storage queue generate-sas', queue_path + 'generate_queue_shared_access_signature', factory)
-cli_storage_data_plane_command('storage queue stats', queue_path + 'get_queue_service_stats', factory)
+
+if get_api_version(ResourceType.MGMT_NETWORK) in ['2016-12-01']:
```
Comment (bganapa) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py]: fixed

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py (hunk 2 above)
Comment (bganapa) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: removed the todo

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py (hunk 5 above)
Comment (bganapa) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: fixed them.. thanks for the review Derek!

**Pull request merged**